src/proto/plumtree.rs

```diff
@@ -363,11 +373,13 @@ pub struct State<PI> {
 
     /// [`Stats`] of this plumtree.
     pub(crate) stats: Stats,
+
+    max_message_size: usize,
 }
 
 impl<PI: PeerIdentity> State<PI> {
     /// Initialize the [`State`] of a plumtree.
-    pub fn new(me: PI, config: Config) -> Self {
+    pub fn new(me: PI, config: Config, max_message_size: usize) -> Self {
```

Comment:
thought on adding to `Config` itself? sounds a lot like a config param

Reply:
Thought so too at first, but the `max_message_size` config setting is shared between `plumtree::Config` and `hyparview::Config`, so adding it to `plumtree::Config` would allow for setting different `max_message_size` on the plumtree level and the upper config level, which is not ideal - that's why I opted for this way. Open to changing it, but not sure what exactly makes more sense.

Reply:
yeah neither is really that much of a better solution